sdk/CONTRIBUTING.md

```diff
@@ -1,12 +1,69 @@
 ## Contributing to the `kfp` SDK
-KFP v2 for SDK is under development on the `master` branch.
 
-To contribute to v1, please go to [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your change.
+For developing KFP v2 for SDK use the `master` branch.
+
+For developing KFP v1 for SDK use the [sdk/release-1.8 branch](https://github.com/kubeflow/pipelines/tree/sdk/release-1.8) to submit your changes.
 
 For general contribution guidelines including pull request conventions, see [pipelines/CONTRIBUTING.md](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md).
 
-### Requirements
-All development requirement versions are pinned in [requirements-dev.txt](https://github.com/kubeflow/pipelines/blob/master/sdk/python/requirements-dev.txt).
+### Pre-requisites 
+
+Clone the repo: 
+
+```bash
+git clone https://github.com/kubeflow/pipelines.git && cd pipelines
+```
+
+We recommend working within a [virtual env](https://docs.python.org/3/library/venv.html) when developing for the SDK.
```

Comment:
I'd recommend specifying some stable Python version for the user to develop with. There can be quite a bit of variance in the stability of different packages.

Reply:
Also, I don't think we recommend `virtual env`.

Reply:
Imho, it is more suitable to say that the user should use a "package manager". There are a lot of options upstream. That is more community-friendly.

Reply:
venv is not a package manager, that would be pi, but more so a tool to create isolated environments, still I agree we should not be opinionated here as there are other tools, I will rephrase

Reply:
```suggestion
You can use [virtual env](https://docs.python.org/3/library/venv.html) when developing the SDK or whatever Python environment manager you prefer.
```

Reply:
posted an update